src/SablierV2MerkleLockupFactory.sol

```diff
@@ -32,8 +118,24 @@ contract SablierV2MerkleLockupFactory is ISablierV2MerkleLockupFactory {
         external
         returns (ISablierV2MerkleLL merkleLL)
     {
-        // Deploy the MerkleLockup contract with CREATE.
-        merkleLL = new SablierV2MerkleLL(baseParams, lockupLinear, streamDurations);
+        // Hash the parameters to generate a salt.
+        bytes32 salt = keccak256(
+            abi.encodePacked(
+                baseParams.asset,
```

Comment:
We also need to include the `msg.sender` to prevent cross-chain replays.

Unlikely to be an issue because the `initialAdmin` is part of the salt, but it shouldn't hurt to add this as an additional precautionary measure.

Reply:
Good point

Reply:
Addressed in https://github.com/sablier-labs/v2-periphery/pull/355/commits/fa7eb777b4b0597af7827c21a92394de8a45e327